### PR TITLE
Py canvasapi

### PR DESCRIPTION
**Pull request by j-hui**
**UNTESTED. do not merge yet, try this out on lab1 or something first.**

Got rid of my poor man's library, using `canvasapi`'s `submissions_bulk_updated()` [endpoint](https://canvasapi.readthedocs.io/en/latest/assignment-ref.html#canvasapi.assignment.Assignment.submissions_bulk_update) now

### CODE REVIEW

**Comment by j-hui** [comment]
addresses #4

File: canvas-wrangler.py [hunk 1]
```diff
@@ -86,88 +85,51 @@ def retrieve_indices(header, opt):
...
 
-    gradesheet = build_gradesheet(grades, student_idx, grade_idx, comment_idx)
-    
-    headers = cvl.auth_header(opt.canvasToken())
-    uri, data = cvl.update_grades(opt.course_id, opt.assignment_id, gradesheet)
-    url = cvl.format_url(uri)
+    grade_data = build_grade_data(grades, student_idx, grade_idx, comment_idx)
 
-    log_dump_wrangler(url, data, headers, opt)
+    canvasURL = 'https://courseworks2.columbia.edu/api/v1' # TODO: don't hard
```
Comment (nsgomez) [inline canvas-wrangler.py:127]: Appending `/api/v1` is deprecated in canvasapi:

>`canvasapi/canvas.py:40: DeprecationWarning: 'base_url' no longer requires an API version be specified. Rewriting 'base_url' to https://courseworks2.columbia.edu`

File: canvas-wrangler.py [hunk 2]
```diff
@@ -86,88 +85,51 @@ def retrieve_indices(header, opt):
...
+    canvas = Canvas(canvasURL, opt.canvasToken())
+    course = canvas.get_course(opt.course_id)
+    assignment = course.get_assignment(opt.assignment_id)
+
+    log_dump_wrangler(data, course, assignment, opt)
```
Comment (nsgomez) [inline canvas-wrangler.py:132]: Is `data` here supposed to be `grade_data`? `data` isn't defined anymore so this throws a NameError.

File: canvas-wrangler.py [hunk 3]
```diff
@@ -176,11 +138,19 @@ def wrangle_canvas(opt):
...
 
-    if cont:
-        opt.log('Submitting to Canvas...')
-        return submit_grades(url, data, headers, opt)
+    if not cont:
+        opt.log('Aborting; not submiting to Canvas')
+        return
+
+    opt.log('Submitting to Canvas...')
+    progress = submit_grades(data, assignment, opt)
```
Comment (nsgomez) [inline canvas-wrangler.py:146]: `data` should be `grade_data`

**Review by nsgomez — APPROVED** [review]
Tested with lab2. After making the changes I suggested, grades posted fine. Checked a dozen or so random UNIs and their comments and grades were correct, so LGTM 👍

**Comment by j-hui** [comment]
huh. how tf did i upload grades for 4118 HW1 🤔

**Comment by nsgomez** [comment]
definitely a big think there

**Comment by j-hui** [comment]
turns out i didn't even make those changes there at all. i just did and manually pushed the changes there. going to merge that one into master, closing this one

**Pull request closed without merging**